pkg/pxc/statefulset.go

```diff
@@ -120,6 +119,13 @@ func StatefulSet(ctx context.Context, cl client.Client, sfs api.StatefulApp, pod
 		}
 	}
 
+	customAnnotations := podSpec.Annotations
+	if cr.CompareVersionWith("1.17.0") >= 0 {
+		customAnnotations = map[string]string{
+			"kubectl.kubernetes.io/default-container": sfs.Labels()[naming.LabelAppKubernetesComponent],
```

Comment:
I think that if the `customAnnotations` already contains annotations, we need to add the `default container` together with all the other without overwriting them. If the `customAnnotations` is `nil`, we should make a new map and add the annotation as we do in these lines.